packages/dds/tree/src/test/simple-tree/getJsonSchema.spec.ts

```diff
@@ -0,0 +1,333 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import { getJsonSchema, SchemaFactory, type TreeJsonSchema } from "../../simple-tree/index.js";
+
+import { hydrate } from "./utils.js";
+import { getJsonValidator } from "./jsonSchemaUtilities.js";
+
+describe.only("getJsonSchema", () => {
+	it("Leaf node", async () => {
+		const schemaFactory = new SchemaFactory("test");
+		const Schema = schemaFactory.string;
+
+		const actual = getJsonSchema(Schema);
+
+		const expected: TreeJsonSchema = {
+			$defs: {
+				"com.fluidframework.leaf.string": {
+					type: "string",
+					_kind: "leaf",
+				},
+			},
+			anyOf: [
+				{
+					$ref: "#/$defs/com.fluidframework.leaf.string",
+				},
+			],
+		};
+		assert.deepEqual(actual, expected);
+
+		// Verify that the generated schema is valid.
+		const validator = getJsonValidator(actual);
+
+		// Verify expected data validation behavior.
+		validator(hydrate(Schema, "Hello world"), true);
+		validator("Hello world", true);
+		validator({}, false);
+		validator([], false);
+	});
+
+	it("Array schema", () => {
+		const schemaFactory = new SchemaFactory("test");
+		class Schema extends schemaFactory.array("array", schemaFactory.string) {}
+
+		const actual = getJsonSchema(Schema);
+
+		const expected: TreeJsonSchema = {
+			$defs: {
+				"test.array": {
+					type: "array",
+					_kind: "array",
+					items: {
+						anyOf: [{ $ref: "#/$defs/com.fluidframework.leaf.string" }],
+					},
+				},
+				"com.fluidframework.leaf.string": {
+					type: "string",
+					_kind: "leaf",
+				},
+			},
+			anyOf: [
+				{
+					$ref: "#/$defs/test.array",
+				},
+			],
+		};
+		assert.deepEqual(actual, expected);
+
+		// Verify that the generated schema is valid.
+		const validator = getJsonValidator(actual);
+
+		// Verify expected data validation behavior.
+		validator(hydrate(Schema, ["Hello", "world"]), false); // TODO: this should work
```

Comment:
The error reported from `ajv` in this case is 
```shell
Error: Data failed validation:
        {"instancePath":"","schemaPath":"#/type","keyword":"type","params":{"type":"array"},"message":"must be array"}
        {"instancePath":"","schemaPath":"#/anyOf","keyword":"anyOf","params":{},"message":"must match a schema in anyOf"}
```

Reply:
@CraigMacomber @noencke  a hydrated array node should satisfy the constraints of a JavaScript array, right? All of the other node schema cases handle hydrated input just fine. Arrays seem to be the odd ones out here.

Reply:
Try using "POJO Mode" by defining your array via `const schema = schemaFactory.array(schemaFactory.string);` rather than via the class syntax. If that doesn't work, then my guess is that we just don't satisfy whatever criteria ajv is using to determine that the object is indeed an array.

Reply:
POJO mode indeed fixes the issue. But it isn't clear to me why the two modes should be different. Shouldn't we expect both to work?

Reply:
In practice, we expect POJO to work and not-POJO to not-work, that is the expected behavior. At least, it is for `Array.isArray` which is what I am guessing the library is using to check for array-ness.

Philosophically, I'm not sure why we can't have this work in both cases though. @CraigMacomber, can you please remind us why we can't e.g. have the class-based array node's prototype descend from `[]`, and thus satisfy the prototype check? And, are there other problems in addition to that?

Reply:
Okay, based on offline discussions, it appears that this is effectively as designed. Our class-based schema nodes are not intended to masquerade as POJO data types, so this scenario is not expected to work. Since these tests are specifically testing compatibility with data coming from external systems, I updated the tests to use POJO mode exclusively.

There is likely further discussion that should be had about the larger implications of the two schema modes, but that is beyond the scope of this PR.